### ISSUE TITLE
Datalabels overlaps each other after the redraw call 

### ISSUE THREAD

**Issue by Basts** [highcharts/highcharts#13742]
#### Expected behaviour

Last properly working version is 8.0.4, so overlapping should work as in this version.

Demo: https://jsfiddle.net/BlackLabel/84rdwv21/

#### Actual behaviour

Data labels overlap each other after the redraw call. 

#### Live demo with steps to reproduce

Demo: https://jsfiddle.net/BlackLabel/pf0hq2t3/ - click the `redraw` button to see the issue.

#### Product version

Highcharts JS v8.1.2 (2020-06-16)

`Internal note`

Bisect led me to [#43fce49134](https://github.com/highcharts/highcharts/commit/43fce49134b27c40e88844c2875a7674e3c79fbd)

**Comment by pawelfus** [comment]
FYI @Izothep

**Comment by Izothep** [comment]
**Internal Note:**

It looks that the issue itself is not introduced in mentioned commit. The issue was existing before and was related to incorrect visibility setting in `dataLabels`. Related commit changed method of showing/hiding `dataLabels` so they are not moved out of the graph but the opacity is changing. In old version, `dataLabels` opacity was similarly set to 1 but the position was outside of the graph.

After some analysis, it looks that the problem might be solved by adding two new lines:
```
label.visibility = 'inherit';
label.placed = true;
```
in initial labels visibility setter:
```
for (i = 0; i < len; i++) {
        label = labels[i];
        if (label) {
            // Mark with initial opacity
            label.oldOpacity = label.opacity;
            label.newOpacity = 1;
            label.absoluteBox = getAbsoluteBox(label);
        }
    }
```

**Comment by drmrbrewer** [comment]
@Izothep looks like you're homing in on a solution.. nice!  Just to mention that this issue arose from my post [here](https://stackoverflow.com/q/62412026/4070848), and although the post is slightly rambly (sorry), it would be good if the solution works for all the examples mentioned in the **EDIT** part of my post... I think they are all probably just variants of the example that @Basts used in the OP above, but just in case!

**Comment by Izothep** [comment]
Hi @drmrbrewer! Thank you for the response and for sending a link to more examples. I have needed to change solution to meet all the use-cases.

Please find some examples with new solution, it would be great if you could send me a confirmation if all charts work correctly on your end:
- https://jsfiddle.net/BlackLabel/84rdwv21/1/ - example provided by @Basts 
- https://jsfiddle.net/BlackLabel/u3ezwvth/
- https://jsfiddle.net/BlackLabel/gtL368ay/
- https://jsfiddle.net/BlackLabel/2x1hdu65/
- https://jsfiddle.net/BlackLabel/nro4x1L0/
- https://jsfiddle.net/BlackLabel/nbt26mrc/

*Internal Node:*

It looks that we need to remove `label.placed = !!newOpacity;` from `overlapping-datalabels` module.

**Comment by drmrbrewer** [comment]
@Izothep yes it does appear to work in all use-cases from my stackoverflow post.  Hopefully in all use-cases!  Thanks for your attention to this issue!

**Comment by Izothep** [comment]
Thank you for such a quick response! We will make some further tests to check if the solution is not breaking anything and then hopefully we will make a final commit with the fix.

**Comment by LukaszKokot** [comment]
Hey there @TorsteinHonsi, I apologise if it's not the right place. I just wanted to know what would be the ETA for this fix to be tagged in 8.1.3?

**Comment by TorsteinHonsi** [comment]
v8.1.3 is expected within a few weeks.

**Comment by denis-chattermill** [comment]
Thank you for the fix! Any news about v8.1.3 release date?

**Comment by pawelfus** [comment]
If everything will go smoothly, next release should see the light of day next week :)
